File: src/main/java/unificado/core/service/cuentanegocio/repository/CuentaNegocioRepository.java
package unificado.core.service.cuentanegocio.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import unificado.core.service.cuentanegocio.model.CuentaNegocio;

import java.math.BigDecimal;
import java.util.List;
import java.util.Optional;

public interface CuentaNegocioRepository extends JpaRepository<CuentaNegocio, Long> {

    List<CuentaNegocio> findAllByNumeroMaestro(BigDecimal numeroMaestro);

    Optional<CuentaNegocio> findByCuentaNegocioId(Long cuentaNegocioId);

}

File: src/main/java/unificado/api/rest/kotlin/repository/CuentaNegocioRepository.kt
package unificado.api.rest.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import unificado.api.rest.kotlin.model.CuentaNegocio
import java.math.BigDecimal
import java.util.Optional

interface CuentaNegocioRepository : JpaRepository<CuentaNegocio, Long> {

    fun findByCuentaNegocioId(cuentaNegocioId: Long): Optional<CuentaNegocio?>?

    fun findAllByNumeroMaestro(numeroMaestro: BigDecimal): List<CuentaNegocio?>?

}